policy_validator.c

```diff
@@ -14,6 +16,15 @@ static inline uint64_t policy_validator_reg_reader(uint32_t reg_num)
     return (uint64_t)policy_validator_hack_env->gpr[reg_num];
 }
 
+static inline uint64_t policy_validator_address_fixer(uint64_t vaddr)
+{
+    CPUClass *cc = CPU_GET_CLASS(policy_validator_hack_cpu_state);
+    uint64_t paddr = (cc->get_phys_page_debug(policy_validator_hack_cpu_state, vaddr)|(vaddr & (qemu_real_host_page_size-1)));
```

Comment:
Might want to split this calculation up to clarify the two steps:  1. get page address and 2.  get page offset.